ci/hdf5-build.sh

```diff
@@ -53,60 +35,28 @@ else
         export CFLAGS="$CFLAGS -arch $CIBW_ARCHS_MACOS"
         export CPPFLAGS="$CPPFLAGS -arch $CIBW_ARCHS_MACOS"
         export CXXFLAGS="$CXXFLAGS -arch $CIBW_ARCHS_MACOS"
-
-        download_unpack_zlib
-
-        pushd "$INPUT/zlib-$ZLIB_VERSION"
-        ./configure                             \
-            --archs="-arch $CIBW_ARCHS_MACOS"   \
-            --prefix="$OUTPUT"
-        make
-        make install
-        popd
-
-        ZLIB_ARG="--with-zlib=$OUTPUT"
-
-        if [[ "$CIBW_ARCHS_MACOS" = "arm64" ]]; then
-            HOST_ARG="--host=aarch64-apple-darwin"
-        fi
     fi
 
-    echo "Building & installing libaec"
-    download_unpack_libaec
-    pushd "$INPUT/libaec-$UNIXY_AEC_VERSION"
-    ./configure            \
-        --prefix="$OUTPUT" \
-        ${HOST_ARG}        \
-        --enable-static=no
-
-    make -j "$NPROC"
-    make install
-    popd
-
     echo "Building & installing hdf5"
     download_unpack_hdf5
-    pushd "$INPUT/hdf5-$UNIXY_HDF5_VERSION"
-    ./configure                        \
-        --prefix="$OUTPUT"             \
-        --with-szlib="$OUTPUT"         \
-        ${HOST_ARG}                    \
-        ${ZLIB_ARG}                    \
-        --enable-build-mode=production \
-        --enable-tools=no              \
-        --enable-tests=no              \
-        --enable-static=no
-
-    # {
-    # work around for crosscompile getting: `./H5detect: Bad CPU type in executable`
-    # so build it on the host architecture instead
-    pushd src
-    ( export ARCHFLAGS= ; make H5detect H5make_libsettings CFLAGS= CCLDFLAGS= CXXFLAGS= CPPFLAGS= )
-    popd
-    # }
 
+    mkdir -p "$OUTPUT/build"
+    pushd "$OUTPUT/build"
+    cmake -G'Unix Makefiles' \
+        -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+        -DBUILD_SHARED_LIBS=OFF \
+        -DHDF5_BUILD_UTILS=OFF \
+        -DHDF5_BUILD_HL_LIB=OFF \
+        -DHDF5_BUILD_EXAMPLES=OFF \
+        -DBUILD_TESTING=OFF \
+        -DHDF5_BUILD_TOOLS=OFF \
+        -DHDF5_ENABLE_SZIP_ENCODING=OFF \
+        -DHDF5_ENABLE_Z_LIB_SUPPORT=OFF \
+        -DCMAKE_INSTALL_PREFIX="$INSTALL" \
+        "$INPUT/CMake-hdf5-$UNIXY_HDF5_VERSION/hdf5-$UNIXY_HDF5_VERSION"
     make -j "$NPROC"
     make install
```

Comment:
```suggestion
    cmake -B "$OUTPUT/build" -G'Unix Makefiles' \
        -DCMAKE_BUILD_TYPE=RelWithDebInfo \
        -DBUILD_SHARED_LIBS=OFF \
        -DHDF5_BUILD_UTILS=OFF \
        -DHDF5_BUILD_HL_LIB=OFF \
        -DHDF5_BUILD_EXAMPLES=OFF \
        -DBUILD_TESTING=OFF \
        -DHDF5_BUILD_TOOLS=OFF \
        -DHDF5_ENABLE_SZIP_ENCODING=OFF \
        -DHDF5_ENABLE_Z_LIB_SUPPORT=OFF \
        -DCMAKE_INSTALL_PREFIX="$INSTALL" \
        -S "$INPUT/CMake-hdf5-$UNIXY_HDF5_VERSION/hdf5-$UNIXY_HDF5_VERSION"
    cmake --build "$OUTPUT/build" -j "$NPROC"
    cmake --install "$OUTPUT/build"
```
Less moving around trees, also more flexible w.r.t. generators.